lib/instrumentation/span.js

```diff
@@ -79,6 +80,29 @@ Span.prototype.end = function (endTime) {
 
   this.ended = true
   this._agent.logger.debug('ended span %o', { span: this.id, parent: this.parentId, trace: this.traceId, name: this.name, type: this.type, subtype: this.subtype, action: this.action })
+
+  // TODO: This is expensive! Consider if there's a way to cache some of this
+  if (this._capturedStackTrace !== null &&
+     (this.duration() / 1000 > this._agent._conf.spanFramesMinDuration)) {
```

Comment:
The new check that says "don't do this work is the span is fast".

Reply:
Looks good.

nit: A comment on the `.duration()` method would eventually be nice to state that it is in milliseconds. One currently has to wade through to lib/instrumentation/timers.js and its `elapsed()` method. :)

IIUC this moves the start of `stackman.callsites` processing from span *start* to span *end*... which looks good to me.

Reply:
> from span start to span end

You understand that correctly.  In order to decide whether or not to do this work based on span duration, we need to move this work to the end.  and 👍 on the comment for `GenericSpan.prototype.duration`